theatre/studio/src/panels/DetailPanel/propEditors/DeterminePropEditor.tsx

```diff
@@ -1,115 +1,140 @@
+    PropConfigByType<K>
+  >
 }
 
-const propEditorByPropType: IPropEditorByPropType = {
-  compound: CompoundPropEditor,
+const simplePropEditorByPropType: ISimplePropEditorByPropType = {
   number: NumberPropEditor,
   string: StringPropEditor,
-  enum: () => <></>,
   boolean: BooleanPropEditor,
   stringLiteral: StringLiteralPropEditor,
   rgba: RgbaPropEditor,
 }
 
-export type IEditablePropertyProps<K extends PropTypeConfig['type']> = {
+type IDetailEditablePropertyProps<K extends PropTypeConfig['type']> = {
   obj: SheetObject
   pointerToProp: Pointer<PropConfigByType<K>['valueType']>
   propConfig: PropConfigByType<K>
 }
 
-type IDeterminePropEditorProps<K extends PropTypeConfig['type']> =
-  IEditablePropertyProps<K> & {
+type IDetailDeterminePropEditorProps<K extends PropTypeConfig['type']> =
+  IDetailEditablePropertyProps<K> & {
     visualIndentation: number
   }
 
-const DeterminePropEditor: React.FC<
-  IDeterminePropEditorProps<PropTypeConfig['type']>
+type IKeyframeDeterminePropEditorProps<K extends PropTypeConfig['type']> = {
+  editingTools: IEditingTools<PropConfigByType<K>['valueType']>
+  propConfig: PropConfigByType<K>
+  keyframeValue: PropConfigByType<K>['valueType']
+  displayLabel?: string
+}
+
+export const DetailDeterminePropEditor: React.VFC<
+  IDetailDeterminePropEditorProps<PropTypeConfig['type']>
 > = (p) => {
   const propConfig =
     p.propConfig ?? getPropTypeByPointer(p.pointerToProp, p.obj)
 
-  const PropEditor = propEditorByPropType[propConfig.type]
-
-  return (
-    <PropEditor
-      obj={p.obj}
-      visualIndentation={p.visualIndentation}
-      // @ts-expect-error This is fine
-      pointerToProp={p.pointerToProp}
-      // @ts-expect-error This is fine
-      propConfig={propConfig}
-    />
-  )
+  if (propConfig.type === 'compound') {
+    return (
+      <DetailCompoundPropEditor
+        obj={p.obj}
+        visualIndentation={p.visualIndentation}
+        pointerToProp={p.pointerToProp}
+        propConfig={propConfig}
+      />
+    )
+  } else if (propConfig.type === 'enum') {
+    // hmm: document how enum gets put into ui (through label?)
+    return <></>
+  } else {
+    const PropEditor = simplePropEditorByPropType[propConfig.type]
+
+    return (
+      <DetailSimplePropEditor
+        SimpleEditorComponent={PropEditor}
+        obj={p.obj}
+        visualIndentation={p.visualIndentation}
+        pointerToProp={p.pointerToProp}
+        propConfig={propConfig}
+      />
+    )
+  }
 }
 
-export default DeterminePropEditor
+const KeyframePropEditorContainer = styled.div`
+  padding: 2px;
+  display: flex;
+  align-items: stretch;
+
+  select {
+    min-width: 100px;
+  }
+`
+
+const KeyframePropLabel = styled.span`
+  font-style: normal;
+  font-weight: 400;
+  font-size: 11px;
+  line-height: 13px;
+  letter-spacing: 0.01em;
+  margin-right: 12px;
+  padding: 8px;
+
+  color: #919191;
+`
+
+export const KeyframeDeterminePropEditor: React.VFC<
```

Comment:
Is this the top of the "inline prop editor logic" calls? If so this may be the appropriate place for this comment suggestion:

 Could you add a comment explaining how this is used, how it relates to `KeyframeSimplePropEditor`, how it relates to `DetailDeterminePropEditor`, and the difference between detail and keyframe prop editors. Please add some `@link`s  or `@see`s to point to the usages and related files.

Reply:
Also how is `theatre/studio/src/panels/DetailPanel/propEditors/utils/IPropEditorFC.ts` related?

Reply:
Did not resolve: there is no comment explaining relationship to `KeyframeSimplePropEditor `. I think it is important to understand that if someone is modifying `KeyframeSimplePropEditor `, they will probably also have to consider modifying `DetailDeterminePropEditor` and to understand why they are separate. Example usage is less important because we can use "find usages".